var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -254,6 +254,13 @@ def determine_b2_options(self, spec, options):
                 'toolset=%s' % self.determine_toolset(spec)
             ])
 
+        # clang is not officially supported for pre-compiled headers
+        # and at least in clang 3.9 still fails to build
+        #   http://www.boost.org/build/doc/html/bbv2/reference/precompiled_headers.html
+        #   https://svn.boost.org/trac/boost/ticket/12496
+        if spec.satisfies('%clang'):
```

Comment:
actually I am not sure if 4.X will work or what kind of three month old trunk the maintainer is referring to.
In case it is fixed, this will work forwardly as soon as a final, stable release of clang 4 is out and we can still add it then. Otherwise, it will be broken again.